.clang-format

```diff
@@ -0,0 +1,80 @@
+﻿---
+BasedOnStyle: Google
+AccessModifierOffset: -1
+AllowShortFunctionsOnASingleLine: Empty
+AllowShortBlocksOnASingleLine:    Empty
+BraceWrapping:
+  AfterCaseLabel:  false
+  AfterClass:      true
+  AfterControlStatement: true
+  AfterEnum:       true
+  AfterFunction:   true
+  AfterNamespace:  true
+  AfterObjCDeclaration: true
+  AfterStruct:     true
+  AfterUnion:      true
+  AfterExternBlock: true
+  BeforeCatch:     true
+  BeforeElse:      true
+  BeforeLambdaBody: true
+  BeforeWhile:     true
+  IndentBraces:    false
+  SplitEmptyFunction: false
+  SplitEmptyRecord: false
+  SplitEmptyNamespace: false
+BreakBeforeBraces: Custom
+ColumnLimit: 100
+PointerAlignment: Right
+SpacesInParentheses: true
+SpacesInSquareBrackets: true
+TabWidth: 4
+UseTab: AlignWithSpaces
```

Comment:
> `UseTab: AlignWithSpaces`

Does that mean every tab will be replaced with spaces?

Reply:
It means it will use tabs to indent, but then will use spaces to align outside these tabs. This will only look correct if your tab width is 4.

This could be our opportunity to settle tabs vs spaces.

Personally, I used to be all tabs, but now I'm down with spaces.

Reply:
Well, “tab for indentation, space for ASCII art” is what I expect. The information the tab provides to the user is so precious to destroy it with spaces, so I would have been against a “replace tabs with space” rule. There can't be any any spacing issue to fear if tabs are only used for indentation and spaces only used for ASCII art. So if that's what this option does, that's good.